### ISSUE TITLE
Add stat for tps

### ISSUE THREAD

**Issue by thepiwo** [aeternity/ae_mdw#821]
in per blocks stats add a transactions per second number

**Comment by marc0olo** [comment]
oh yeah - we also plan to track the max. value right?

**Comment by thepiwo** [comment]
sounds good to me (cc @sborrazas )

**Comment by sborrazas** [comment]
Since we only need the max, we're good enough if we add a new endpoint tracking these sort of values:
```
$ curl http://mainnet.aeternity.com/mdw/v2/stats
> {"max_txs_per_second": 123}
```

**Comment by thepiwo** [comment]
@sborrazas can we also expose at what block that was recoded there?

**Comment by sborrazas** [comment]
> @sborrazas can we also expose at what block that was recoded there?

will do!
```
{
  "max_transactions_per_second": 123,
  "max_transactions_per_second_kbi": 456
  "max_transactions_per_second_mbi": 789
 }
```

**Comment by thepiwo** [comment]
very nice

**Comment by janmichek** [comment]
> http://mainnet.aeternity.com/mdw/v2/stats

Looks good. I just think that the correct URL should be.

`http://mainnet.aeternity.io/mdw/v2/totalstats` ?

 Just adding a new attribute should do the work. Or is it necessary to have a new endpoint?

**Comment by janmichek** [comment]
Got it. Thanks for explanation!

**Comment by thepiwo** [comment]
@sborrazas maybe lets reiterate this, as I understood, our current implementation gives the peak number at the border between 2 key-blocks.

I'd propose we only consider micro-block timing and included transaction within one key-block. Specifically all transactions between first and last micro-block if > 1. Within micro-blocks the time is under consensus, thus we could achieve some more reliable numbers with that approach

**Comment by marc0olo** [comment]
I like that proposal! cc @hanssv :-)

**Comment by hanssv** [comment]
Generations shorter than 3 seconds should be very rare, so not including them sounds good. But still, what to present really, really depends on what you want the number to say. With this definition the "optimal" would be key-block > (full) micro-block > (full) micro-block > key-block (where the generation takes 3.001s  😅). That should also be very rare... And it could in theory give a number almost x2 what we can sustain long term. Maybe compute Txs per (wall clock) minute  or something?!

In my world, since there is 20+ different transactions (with different weight/complexity) Tx/s is not a very interesting concept - but I can see someone with little understanding looking for the number to feel comfortable...

**Comment by thepiwo** [comment]
@hanssv I agree on different transaction types, that would be something we could tackle in the future eventually
